From 1a5b9b471892f9a7bbc4877e21a96aab131b0d21 Mon Sep 17 00:00:00 2001
From: Petr Oros <poros@redhat.com>
Date: Thu, 18 Dec 2014 11:58:55 -0500
Subject: [redhat] kabi: Adding symbol __list_add (lib/list_debug.c)

Message-id: <94e0cd464bd8c4501308254a2b2727195982272c.1418903403.git.poros@redhat.com>
Patchwork-id: 102234
O-Subject: [PATCH RHEL7.1 03/48] kabi: Adding symbol __list_add (lib/list_debug.c)
Bugzilla: 1087454 1160751 1160754
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: 1160751,1160754,1160760
    https://bugzilla.redhat.com/show_bug.cgi?id=1160751
    https://bugzilla.redhat.com/show_bug.cgi?id=1160754
    https://bugzilla.redhat.com/show_bug.cgi?id=1160760

    Description:
    ============
    Not called directly
    These are in lib/list_debug.c; the file is unchanged
    between rhel7 and upstream. Note __list_del_entry and list_del are already
    whitelisted.
    Used in ioc4_extint.ko, superpages.ko, xpmem.ko, extint.ko, hwperf.ko, kbar.ko, numatools.ko, uli.ko

Symbol History
==============
 Request:
 ========
 [PATCH RHEL7 BZ726863 158/329] kabi: Adding symbol __list_add
 Jiri Olsa <jolsa@redhat.com> (Sun, 26 Jan 2014 20:13:51 +0100)
 Symbols is already present in RHEL6 and requested again in RHEL7.

 Reviews:
 ========
 N/A

 Request:
 ========
 [PATCH RHEL7 132/417] kabi: Adding symbol __list_add (include/linux/list.h)
 Jiri Olsa <jolsa@redhat.com> (Wed, 12 Feb 2014 15:50:52 +0100)
 Description:
 ============
 To use the struct list_head.

 Reviews:
 ========
 Re: [PATCH RHEL7 132/417] kabi: Adding symbol __list_add (include/linux/list.h)
 Don Zickus <dzickus@redhat.com> (Mon, 17 Mar 2014 16:10:15 -0400)
 Most likely won't change as it affects everything.  Need the internal
 representation because list_add is an inline.
 Acked-by: Don Zickus <dzickus@redhat.com>

 Request:
 ========
 [PATCH RHEL7.1 BZ1087454 02/33] kabi: Adding symbol __list_add (lib/list_debug.c)
 Petr Oros <poros@redhat.com> (Wed, 17 Sep 2014 18:24:48 +0200)
 Description:
 ============
 Add an entry to a generic kernel linked list.
 Used to populate TIPC subscription lists,
 and add entries to the TIPC name table.

 Reviews:
 ========
 Re: [PATCH RHEL7.1 BZ1087454 02/33] kabi: Adding symbol __list_add (lib/list_debug.c)
 Prarit Bhargava <prarit@redhat.com> (Thu, 18 Sep 2014 07:48:44 -0400)
 list_add() is the proper wrapper here IMO.
 Nacked-by: Prarit Bhargava <prarit@redhat.com>
 P.

 Re: [PATCH RHEL7.1 BZ1087454 02/33] kabi: Adding symbol __list_add (lib/list_debug.c)
 Don Zickus <dzickus@redhat.com> (Mon, 27 Oct 2014 09:56:08 -0400)
 But list_add is a static inline wrapper around __list_add, so that isn't
 possible. :-)
 Acked-by: Don Zickus <dzickus@redhat.com>

Signed-off-by: Petr Oros <poros@redhat.com>
---
 create mode 100644 redhat/kabi/kabi-module/kabi_ppc64/__list_add
 create mode 100644 redhat/kabi/kabi-module/kabi_ppc64le/__list_add
 create mode 100644 redhat/kabi/kabi-module/kabi_s390x/__list_add
 create mode 100644 redhat/kabi/kabi-module/kabi_x86_64/__list_add

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/kabi/kabi-module/kabi_ppc64/__list_add b/redhat/kabi/kabi-module/kabi_ppc64/__list_add
new file mode 100644
index 0000000..f8a2c9f
--- /dev/null
+++ b/redhat/kabi/kabi-module/kabi_ppc64/__list_add
@@ -0,0 +1,2 @@
+#1-
+0x0343a1a8 __list_add vmlinux EXPORT_SYMBOL
diff --git a/redhat/kabi/kabi-module/kabi_ppc64le/__list_add b/redhat/kabi/kabi-module/kabi_ppc64le/__list_add
new file mode 100644
index 0000000..f8a2c9f
--- /dev/null
+++ b/redhat/kabi/kabi-module/kabi_ppc64le/__list_add
@@ -0,0 +1,2 @@
+#1-
+0x0343a1a8 __list_add vmlinux EXPORT_SYMBOL
diff --git a/redhat/kabi/kabi-module/kabi_s390x/__list_add b/redhat/kabi/kabi-module/kabi_s390x/__list_add
new file mode 100644
index 0000000..f8a2c9f
--- /dev/null
+++ b/redhat/kabi/kabi-module/kabi_s390x/__list_add
@@ -0,0 +1,2 @@
+#1-
+0x0343a1a8 __list_add vmlinux EXPORT_SYMBOL
diff --git a/redhat/kabi/kabi-module/kabi_x86_64/__list_add b/redhat/kabi/kabi-module/kabi_x86_64/__list_add
new file mode 100644
index 0000000..f8a2c9f
--- /dev/null
+++ b/redhat/kabi/kabi-module/kabi_x86_64/__list_add
@@ -0,0 +1,2 @@
+#1-
+0x0343a1a8 __list_add vmlinux EXPORT_SYMBOL
-- 
1.7.1